src/server/server_family.cc

```diff
@@ -638,6 +638,15 @@ optional<ReplicaOfArgs> ReplicaOfArgs::FromCmdArgs(CmdArgList args, SinkReplyBui
   return replicaof_args;
 }
 
+uint64_t GetDelayMs(uint64_t ts) {
+  uint64_t now_ns = fb2::ProactorBase::GetMonotonicTimeNs();
+  uint64_t delay_ns = 0;
+  if (ts < now_ns - 1000000) {  // handle states where ts is larger than now
+    delay_ns = (now_ns - ts) / 1000000;
+  }
```

Comment:
Why would `ts` ever be larger than `now_ns`, and even if it is, why would you compare that to now - 1s? and why do you set the result to be in units of seconds? and if now is smaller, wouldn't `now_ns - ts` be a negative (i.e. huge unsigned) number?

Reply:
because oldest_pending_send_ts can be INT64_MAX (its default value). I compare with `now-1ms` just because it's my resolution scale - we do not really care about send being stuck for less than 1ms. now can not be smaller than 1ms assuming that linux monotonic clock did not reset.

Reply:
Sorry, I meant to remove my comment and only keep the modified suggestions. I understood it when I made the suggestion :(